src/providers/ldap/ldap_common.c

```diff
@@ -158,32 +158,30 @@ static void sdap_finalize(struct tevent_context *ev,
                           void *siginfo,
                           void *private_data)
 {
+    struct sdap_id_ctx *id_ctx;
+
+    id_ctx = talloc_get_type(private_data, struct sdap_id_ctx);
+    if (id_ctx != NULL && id_ctx->conn != NULL) {
+        talloc_zfree(id_ctx->conn);
+    }
+
     orderly_shutdown(0);
 }
 
-errno_t sdap_install_sigterm_handler(TALLOC_CTX *mem_ctx,
+errno_t sdap_install_sigterm_handler(struct sdap_id_ctx *mem_ctx,
```

Comment:
Hi,

can you rename the variable as well, I think `mem_ctx` might be misleading, better would be `id_ctx` or `sdap_id_ctx`.

Reply:
updated